### PR TITLE
[Enhancement] Add ICEBERG.MetricsConfig in query profile for Iceberg tables

### PR DESCRIPTION
**Pull request by lipenglin**
Why I'm doing:
To optimize the MetricsModes of the Iceberg tables, it's necessary to display the columns of the `FULL` or `TRUNCATE` type MetricsMode in the ICEBERG query profile.

What I'm doing:
e.g.
```
     - DeployDataSize: 3615
    ICEBERG:
      ComplexMetricsConfig:
        hive_prod:
          testdb:
            census:
               - size: 3
               - columns: {census_year=truncate(16), year=truncate(16), name=truncate(16)}
      ScanMetrics:
        hive_prod:
```

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -550,6 +558,33 @@ private void collectTableStatisticsAndCacheIcebergSplit(Table table, ScalarOpera
...
+     * Under this mode, value_counts, null_value_counts, nan_value_counts and full lower_bounds,
+     * upper_bounds are persisted.
+     * </p>
+     */
+    private static Map<String, MetricsModes.MetricsMode> getComplexIcebergMetricsConfig(IcebergTable table) {
+        MetricsConfig metricsConf = MetricsConfig.forTable(table.getNativeTable());
+        Map<String, MetricsModes.MetricsMode> filedToMetricsMode = Maps.newHashMap();
+        for (Types.NestedField field : table.getNativeTable().schema().columns()) {
+            MetricsModes.MetricsMode mode = metricsConf.columnMode(field.name());
+            if (mode.equals(MetricsModes.Full.get()) || mode.toString().toLowerCase().startsWith("truncate")) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:581]: The default mode is `truncate(16)`. I think we don't need to record the default mode columns

**Comment by Youngwb** [comment]
could you add some test to cover this?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 1 above)
Comment (lipenglin) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java]: done

**Comment by lipenglin** [comment]
> could you add some test to cover this?

done

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by lipenglin** [comment]
Fix conflicts caused by pull rebase.

**Review by Youngwb — APPROVED** [review]

**Comment by lipenglin** [comment]
@stephen-shelby PTAL.
Fix conflicts by `pull rebase`.

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**